lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx

```diff
@@ -4,225 +4,203 @@
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('adds additional specified attributes to the div', function() {
-    const component = getShallow({ id: 'foo', title: 'bar' });
+  it('adds additional specified attributes', function() {
+    const el = renderEl({ id: 'foo', title: 'bar' });
 
-    expect(component).toHaveProp('id', 'foo');
-    expect(component).toHaveProp('title', 'bar');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('title', 'bar');
   });
 
-  it('adds a specified ref to the div', function() {
-    const ref = React.createRef<HTMLDivElement>();
-    const component = mount(<><NxSearchDropdown ref={ref} { ...minimalProps } /></>);
+  it('adds a specified ref', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        renderedEl = renderEl({ ref });
 
-    expect(ref.current).toBeDefined();
-    expect(component.getDOMNode()).toBe(ref.current);
+    expect(ref.current).toBe(renderedEl);
   });
 
-  it('has an NxFilterInput child with the nx-search-dropdown__input class and searchbox role', function() {
-    const component = getMounted(),
-        filterInput = component.find(NxFilterInput),
-        nativeInput = filterInput.find('input');
+  it('has an input field with role="searchbox"', function() {
+    const el = quickRender(),
+        input = el.getByRole('searchbox');
 
-    expect(filterInput).toExist();
-    expect(filterInput).toHaveClassName('nx-search-dropdown__input');
-    expect(nativeInput).toHaveProp('role', 'searchbox');
+    expect(input).toBeInTheDocument();
   });
 
   it('sets the searchText as the value of the input', function() {
-    const component = getShallow({ searchText: 'foo' }),
-        input = component.find(NxFilterInput);
+    const el = quickRender({ searchText: 'foo' }),
+        input = el.getByRole('searchbox');
 
-    expect(input).toHaveProp('value', 'foo');
+    expect(input).toHaveAttribute('value', 'foo');
   });
 
-  it('calls onSearchTextChange whenver the input\'s onChange event fires', function() {
+  it('calls onSearchTextChange whenver the input\'s onChange event fires', async function() {
     const onSearchTextChange = jest.fn(),
-        component = getShallow({ searchText: 'foo', onSearchTextChange }),
-        input = component.find(NxFilterInput);
+        el = quickRender({ searchText: '', onSearchTextChange }),
+        input = el.getByRole('searchbox');
 
     expect(onSearchTextChange).not.toHaveBeenCalled();
 
-    input.simulate('change', 'asdf');
-
-    expect(onSearchTextChange).toHaveBeenCalledWith('asdf');
+    fireEvent.change(input, { target: { value: 'aasd' }});
+    expect(onSearchTextChange).toHaveBeenCalledWith('aasd');
   });
 
-  it('calls onSearch whenver the input\'s onChange event fires with a value that differs after trimming, ' +
-      'passing the trimmed value', function() {
+  it('calls onSearch whenever the input\'s onChange event fires with a value that differs after trimming, ' +
```

Comment:
Also should be in terms of user events and use `userEvent`.

Reply:
> Also should be in terms of user events and use `userEvent`. I _think_ the `initalSelectionStart` parameter on `userEvent.type` can be used to enter text (spaces for instance) in the input at positions other than the end.

I've had a lot of difficulty using `userEvent` in this component. Not sure if I'm not implementing it correctly, but I can't seem to change the input field with something like `user.type(input, '{backspace}'`. I also couldn't find documentation about  `initalSelectionStart`, so I tried using something like `user.type(input, '{arrowleft>4/} a'` to make 4 arrow left key presses to modify the start of the text, but it doesn't seem to work. I might need some help on this.

Reply:
> I've had a lot of difficulty using userEvent in this component. Not sure if I'm not implementing it correctly, but I can't seem to change the input field with something like user.type(input, '{backspace}'. I also couldn't find documentation about initalSelectionStart, so I tried using something like user.type(input, '{arrowleft>4/} a' to make 4 arrow left key presses to modify the start of the text, but it doesn't seem to work. I might need some help on this.

Did you try `user.keyboard`? From the [user.type docs](https://testing-library.com/docs/user-event/utility#type):
 
> You should use [keyboard()](https://testing-library.com/docs/user-event/keyboard) if you want to just simulate pressing buttons on the keyboard.
> You can use type() if you just want to conveniently insert some text into an input field or textarea.

That said, initialSelectionStart` is only a parameter on `type`, and not on `keyboard`. So you'd have to figure out a different way for that part.  I'll look into it

Reply:
> That said, initialSelectionStart is only a parameter on type, and not on keyboard. So you'd have to figure out a different way for that part. I'll look into it

See 0edb8b1bf